sdk/storage/azdatalake/directory/client.go

```diff
@@ -123,9 +159,14 @@ func (d *Client) sharedKey() *exported.SharedKeyCredential {
 	return base.SharedKeyComposite((*base.CompositeClient[generated.PathClient, generated.PathClient, blob.Client])(d))
 }
 
-// URL returns the URL endpoint used by the Client object.
-func (d *Client) URL() string {
-	return "s.generated().Endpoint()"
+// DFSURL returns the URL endpoint used by the Client object.
+func (d *Client) DFSURL() string {
```

Comment:
is there a reason we want to expose both URL types?

Reply:
yea, the reason being that we have apis that use generated datalake code but need the blob endpoint rather than dfs, for example undelete_path